modular_skyrat/modules/morenarcotics/code/reagents/meth.dm

```diff
@@ -0,0 +1,169 @@
+	overdose_threshold = 20
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	ph = 5
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	addiction_types = list(/datum/addiction/stimulants = 24) // i know im gonna get shit for tweaking meth so ill just set this to something modest
+
+/datum/reagent/drug/bluesky/on_new(data)
+	. = ..()
+	var/effective_impurity = min(1, (1 - creation_purity)/0.5)
+	color = BlendRGB(initial(color), "#FAFAFA", effective_impurity)
+
+/datum/reagent/drug/bluesky/on_merge(data, amount)
+	. = ..()
+	var/effective_impurity = min(1, (1 - creation_purity)/0.5)
+	color = BlendRGB(initial(color), "#FAFAFA", effective_impurity)
+
+/datum/reagent/drug/bluesky/on_mob_metabolize(mob/living/L)
+	..()
+	L.add_movespeed_modifier(/datum/movespeed_modifier/reagent/methamphetamine)
+
+/datum/reagent/drug/bluesky/on_mob_end_metabolize(mob/living/L)
+	L.remove_movespeed_modifier(/datum/movespeed_modifier/reagent/methamphetamine)
+	..()
+
+/datum/reagent/drug/bluesky/on_mob_life(mob/living/carbon/affected_mob, seconds_per_tick, times_fired)
+	var/high_message = pick("You feel hyper.", "You feel like you need to go faster.", "You feel like you can run the world.")
+	if(SPT_PROB(2.5, seconds_per_tick))
+		to_chat(affected_mob, span_notice("[high_message]"))
+	var/strength_multiplier = creation_purity**3
+	affected_mob.add_mood_event("tweaking", /datum/mood_event/stimulant_medium, name)
+	affected_mob.AdjustStun(-35 * strength_multiplier * REM * seconds_per_tick)
+	affected_mob.AdjustKnockdown(-35 * strength_multiplier * REM * seconds_per_tick)
+	affected_mob.AdjustUnconscious(-35 * strength_multiplier * REM * seconds_per_tick)
+	affected_mob.AdjustParalyzed(-35 * strength_multiplier * REM * seconds_per_tick)
+	affected_mob.AdjustImmobilized(-35 * strength_multiplier * REM * seconds_per_tick)
+	affected_mob.adjustStaminaLoss(-2 * strength_multiplier * REM * seconds_per_tick, FALSE, required_biotype = affected_biotype)
+	affected_mob.set_jitter_if_lower(4 SECONDS * REM * seconds_per_tick)
+	affected_mob.adjustOrganLoss(ORGAN_SLOT_BRAIN, rand(2, 5) * REM * seconds_per_tick) // more brain damage than usual
+	if(SPT_PROB(2.5, seconds_per_tick))
+		affected_mob.emote(pick("twitch", "shiver"))
+	..()
+	. = TRUE
+
+/datum/reagent/drug/bluesky/overdose_process(mob/living/affected_mob, seconds_per_tick, times_fired)
+	if(!HAS_TRAIT(affected_mob, TRAIT_IMMOBILIZED) && !ismovable(affected_mob.loc))
+		for(var/i in 1 to round(4 * REM * seconds_per_tick, 1))
+			step(affected_mob, pick(GLOB.cardinals))
+	if(SPT_PROB(10, seconds_per_tick))
+		affected_mob.emote("laugh")
+	if(SPT_PROB(18, seconds_per_tick))
+		affected_mob.visible_message(span_danger("[affected_mob]'s hands flip out and flail everywhere!"))
+		affected_mob.drop_all_held_items()
+	..()
+	affected_mob.adjustToxLoss(1 * REM * seconds_per_tick, FALSE, required_biotype = affected_biotype)
+	affected_mob.adjustOrganLoss(ORGAN_SLOT_BRAIN, (rand(5, 10) / 10) * REM * seconds_per_tick)
```

Comment:
I'm tired of repeating, please just use defines instead of magic numbers that mean nothing everywhere.